ouroboros-consensus-test/src/Test/Ouroboros/Consensus/ChainGenerator/RaceIterator.hs

```diff
@@ -0,0 +1,136 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE PatternSynonyms #-}
+{-# LANGUAGE PolyKinds #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+
+{-# OPTIONS_GHC -Wno-unticked-promoted-constructors #-}
+
+module Test.Ouroboros.Consensus.ChainGenerator.RaceIterator (
+  RaceLbl,
+  Race (Race, UnsafeRace),
```

Comment:
I  intentionally export `Unsafe*` in general; it's the user's fault for using it without understanding it.

I think it's worse to make it more difficult for someone to explore some additional behavior that would otherwise necessitate patching our code.